crypto/ocsp/ocsp_lib.c

```diff
@@ -79,8 +79,11 @@ OCSP_CERTID *OCSP_cert_id_new(const EVP_MD *dgst, const X509_NAME *issuerName,
   if (!(ASN1_OCTET_STRING_set(cid->issuerKeyHash, md, i))) {
     goto err;
   }
-  if (ASN1_STRING_copy(cid->serialNumber, serialNumber) == 0) {
-    goto err;
+  // Only copy |serialNumber| if available. This may be empty.
```

Comment:
In this case, it was because `OCSP_cert_to_id` allows for a `NULL` subject, then passes the `NULL` `serialNumber` to `OCSP_cert_id_new`.  OpenSSL allows the `NULL` parameter and just simply skips populating it. 
This was intended for self-signed certificates in `OCSP_cert_to_id` [extracts](https://github.com/aws/aws-lc/blob/af35ef1f8b4232082d19a887a6df9d6401b7fbf8/crypto/ocsp/ocsp_lib.c#L27C13-L27C34)  where the subject name is extracted from the issuer and a serial number is given `NULL`.